src/LightspeedRetailClient.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace ZfrLightspeedRetail;
+
+use GuzzleHttp\Command\CommandInterface;
+use GuzzleHttp\Command\ResultInterface;
+use GuzzleHttp\Command\ServiceClientInterface;
+use Psr\Http\Message\ResponseInterface;
+use Traversable;
+
+/**
+ * HTTP Client used to interact with the Lightspeed Retail API
+ *
+ * @author Daniel Gimenes
+ * @author Michaël Gallego
+ *
+ * CUSTOMER RELATED METHODS:
+ *
+ * @method array getCustomers(array $args = [])
+ * @method array createCustomer(array $args = [])
+ * @method array updateCustomer(array $args = [])
+ *
+ * ITEM RELATED METHODS:
+ *
+ * @method array getItems(array $args = [])
+ * @method array createItem(array $args = [])
+ * @method array updateItem(array $args = [])
+ *
+ * SALE RELATED METHODS:
+ *
+ * @method array getSales(array $args = [])
+ *
+ * ITERATOR METHODS:
+ *
+ * @method Traversable getSalesIterator(array $args = [])
+ */
+class LightspeedRetailClient
+{
+    /**
+     * @var ServiceClientInterface
+     */
+    private $serviceClient;
+
+    /**
+     * @param ServiceClientInterface $serviceClient
+     */
+    public function __construct(ServiceClientInterface $serviceClient)
```

Comment:
Can't you make it nullable and create a default one like I do here: https://github.com/zf-fr/zfr-tradegecko/blob/master/src/TradeGeckoClient.php#L326 Makes usage must ismpler and allows to re-create a new client without knowing all the internal details such as the path of the descriptor... I think this should be part of the internal of this client. The injectioon is mostly here only for testing purposes.